transaction/std/src/memo/gift_code_sender.rs

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Object for 0x0002 Gift Code Sender memo type
+//!
+//! This was proposed for standardization in mobilecoinfoundation/mcips/pull/32
+
+use crate::{impl_memo_type_conversions, RegisteredMemoType};
+use mc_transaction_core::MemoError;
+use std::{convert::TryFrom, str};
+
+/// A gift code is considered "redeemed" when the receiver of
+/// a gift code message uses the private spend key of the gift
+/// code TxOut (originally sent to the sender's reserved gift
+/// subaddress) to send the TxOut to them themselves. When that
+/// happens, the receiver can write an optional 64 byte null
+/// terminated utf-8 string in the memo field of an associated
+/// change TxOut to record any desired info about the gift code
+/// redemption.
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub struct GiftCodeSenderMemo {
+    /// The data representing the gift code memo
+    memo_data: [u8; Self::MEMO_DATA_LEN],
+}
+
+impl RegisteredMemoType for GiftCodeSenderMemo {
+    const MEMO_TYPE_BYTES: [u8; 2] = [0x00, 0x02];
+}
+
+impl GiftCodeSenderMemo {
+    /// The length of the custom memo data.
+    pub const MEMO_DATA_LEN: usize = 64;
+
+    /// Create a new gift code memo
+    pub fn new(note_data: &'static str) -> Result<Self, MemoError> {
```

Comment:
What do I do if I want to attach a memo that is not a static string, e.g. something that came from a `String`? This should be `&str`

Reply:
I'm also not sure we need both `new` and a TryFrom